ray-operator/test/sampleyaml/rayjob_test.go

```diff
@@ -65,9 +67,7 @@ func TestRayJob(t *testing.T) {
 					desiredWorkerReplicas += *workerGroupSpec.Replicas
 				}
 			}
-
 			g.Eventually(WorkerPods(test, rayCluster), TestTimeoutShort).Should(HaveLen(int(desiredWorkerReplicas)))
-			g.Expect(rayCluster.Status.DesiredWorkerReplicas).To(Equal(desiredWorkerReplicas))
```

Comment:
why remove this?

Reply:
I think we don't need to check this because we've already checked the number of worker pods above. Also because this is a one-line fix so I do it directly in this refactoring PR. Do you think it's better to create a separate PR?

Reply:
If I understand correctly, this differs from the above logic. The above logic checks the number of worker Pods by querying the K8s API server, which is the source of truth. The deleted line checks whether the RayCluster status reflects the actual cluster status.

Reply:
Okay I've restored this check, by updating it to 

```go
g.Expect(GetRayCluster(test, namespace.Name, rayCluster.Name)).To(WithTransform(RayClusterDesiredWorkerReplicas, Equal(desiredWorkerReplicas)))
```

because I found that we forgot to get the RayCluster again to update its status.

Reply:
Using a transform just to return a field seems overkill :P

Reply:
I used `WithTransform` because I found there already been a `RayClusterDesiredWorkerReplicas` function.

Reply:
And `GetRayCluster` returns 2 parameters: one is `raycluster`, and the other is `error`. `WithTransform` can successfully take only the first one as input and fail if `err` is not `nil`. I think this is an ideal and neat syntax.